Directory.Packages.props

```diff
@@ -23,7 +23,17 @@
     <PackageVersion Include="Microsoft.Extensions.DependencyInjection" Version="[3.1.0,)" />
     <PackageVersion Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="[3.1.0,)" />
     <PackageVersion Include="Microsoft.Extensions.Hosting.Abstractions" Version="[2.1.0,)" />
-    <PackageVersion Include="Microsoft.Extensions.Logging" Version="[3.1.0,)" />
+
+    <!--
+        Typically, the latest stable version of Microsoft.Extensions.Logging should be used here because:
+        1) Each major version bump will have some new API capabilities (e.g. .NET 6 introduced Compile-time logging
+          source generation, .NET 8 introduced automatic event id generation).
+        2) Each minor version bump is normally security hotfixes or critical bug fixes.
+        3) Since version 3.1.0, the .NET runtime team provides extra backward compatibility guarantee to
+          Microsoft.Extensions.Logging even during major version bumps, so compatibility is not a concern here.
+    -->
+    <PackageVersion Include="Microsoft.Extensions.Logging" Version="8.0.0-rc.1.23419.4" />
```

Comment:
Please add changelog entry for this.

Reply:
I am on the fence, since this change is not supposed to break the user at all, do we want to mention it?

Reply:
I guess it's okay to add this considering as a new feature? E.g. we did the same for https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Api/CHANGELOG.md#160-alpha1.

Reply:
Its still bumping the dependency multiple major versions. Users in some environments like Azure Functions would definitely want to know which version we bumped this, so they can stay in older OTel sdks. Changelog is the best place to communicate this.

SD.DS updates were always added in changelogs : https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Api/CHANGELOG.md#140-rc1

Reply:
I've added a changelog entry, it might be a bit weird since the version says "8.0.0-rc.1.23419.4", for the actual stable release we might want to add another changelog entry or modify existing one (doesn't seem to be great either way).